### DISCUSSION TITLE
Raspberry Pi 4 + Ubuntu 24.04 LTS

### DISCUSSION THREAD

**Discussion by hendymen** [OpenSurv/OpenSurv#7] [Hardware + Operating system]
All experiences and tests with OpenSurv on Raspberry Pi 4 and Ubuntu 24.04 LTS

Running 3* 720P H.265 RTSP streams on RPI 4 / 8GB RAM / Ubuntu 24.04 without any problems.
 
![Afbeelding van WhatsApp op 2024-11-12 om 21 00 11_c378d5c8](https://github.com/user-attachments/assets/06c07794-de4a-4486-a868-face306fccda)
(picture is from playback of local mp4 file, but 3* 720P H.265 RTSP running great!)

Also tries all steams to be native, however these are 4K/8mp cameras, connection took a while and picture was directly out of sync. Lack of performance in RPI4. 

However 3* 720P is also maxing out the amount of pixels on this monitor.

**Comment by SvenVD** [comment]
Nice, thanks for reporting back!

**Comment by jprikeasc** [comment]
H264 streams seem slow - what can I do beyond the TCP and other troubleshooting steps.

**Comment by KPMCA** [comment]
On Rasperry Pi 4 Model B Rev 1.5 running raspbian lite, I'm getting the default lightdm desktop login screen. I tried the full version with similar problem.

![opensurv_rpi4_raspbianlite](https://github.com/user-attachments/assets/593899b6-26cc-4989-9db5-3d6c003f9e57)

Logging in takes me to the lightdm desktop but doesn't show the demo streams.

**Reply by SvenVD** [reply]
Others above reported using Ubuntu 24.04 on the raspberry pi. Maybe better chance to try that one

**Comment by jprikeasc** [comment]
Had issues with latency on an rtsp stream. I used ffmpeg and it worked .

JP Rike
Chief Technology Officer

Work: +12028976460

Mobile: +17037272330

Email: ***@***.***

Website: www.asc-defense.com
This e-mail may contain confidential information. It is intended solely for the addressee. Access to this e-mail by anyone else is unauthorized. If you are not the intended recipient, you must not use, copy, disclose, or take any action based on this e-mail or any information herein. If you have received this e-mail in error, please notify the sender and permanently delete the e-mail and any attachments immediately.
---------------------------------------------------------------

**Reply by SvenVD** [reply]
> Had issues with latency on an rtsp stream. I used ffmpeg and it worked 

To reduce latency you can try this https://github.com/OpenSurv/OpenSurv/blob/4a8f282b2aa747dd5c5999a754336a29884563c6/README.md?plain=1#L130

**Comment by jprikeasc** [comment]
Unfortunately, that did not work, and neither did TCP.

Even reduced the stream to 640x480 and 5 frames with no luck.

JP Rike
Chief Technology Officer

Work: +12028976460

Mobile: +17037272330

Email: ***@***.***

Website: www.asc-defense.com
This e-mail may contain confidential information. It is intended solely for the addressee. Access to this e-mail by anyone else is unauthorized. If you are not the intended recipient, you must not use, copy, disclose, or take any action based on this e-mail or any information herein. If you have received this e-mail in error, please notify the sender and permanently delete the e-mail and any attachments immediately.
---------------------------------------------------------------

**Comment by UntouchedWagons** [comment]
Yup I've got opensurv working on a Pi 4 with the 7" touchscreen and Ubuntu 24.04. I'm going to see if I can get opensurv to work on DietPi, Ubuntu is way too bloated for this use case.
